llvm/tools/llvm-profgen/PerfReader.cpp

```diff
@@ -375,6 +376,9 @@ PerfScriptReader::convertPerfDataToTrace(ProfiledBinary *Binary,
                                           StringRef(ErrorFile)};    // Stderr
   sys::ExecuteAndWait(PerfPath, ScriptMMapArgs, std::nullopt, Redirects);
 
+  PerfScriptReader::markTempFile(PerfTraceFile);
+  PerfScriptReader::markTempFile(ErrorFile);
```

Comment:
nit: I would slightly prefer we do "createTempFile" instead of "createFile" + "MarkTemp". no strong opinion though.

Reply:
I don't think `sys::fs::createTemporaryFile` has any of the cleanup logic from `MarkTemp`. There is the `TempFile` class but as far as I can tell that is meant towards the use case of getting a C++ file descriptor that you can write to but doesn't work to give you a filename + later cleanup; you only get a filename when you explicitely do not want the cleanup to happen.

So while I looked a bit when reviewing, I don't think there is much better than can be done here?

Reply:
I meant to suggest combing "createUniquePath" and "markTempFile" into a single helper, so people won't create a temp file and forget to mark it as temp..